main.go

```diff
@@ -40,8 +40,13 @@ func printStartUpMessage() {
 }
 
 type TestRunner interface {
+	// Take testCases as input, mutate runner.RunResult -> adding test results to it.
```

Comment:
[nit] Most go doc starts with the name of the function/attributes that it explains.
```suggestion
	// Run takes testCases as input, executes the test against the test cases, and updates the runner.RunResult with the test results.
```